sky/backends/backend_utils.py

```diff
@@ -1768,8 +1768,15 @@ def check_owner_identity(cluster_name: str) -> None:
                         f'The cluster was owned by {owner_identity}, but '
                         f'a new identity {current_user_identity} is activated. We still '
                         'allow the operation as the two identities are likely to have '
-                        'the same access to the cluster, but please be aware that '
-                        'this is not guaranteed.')
+                        'the same access to the cluster. Please be aware that this can '
+                        'cause unexpected cluster leakage if the two identites have '
+                        'different access to the cluster.')
```

Comment:
How about “if the two identities are not actually equivalent (e.g., belong to the same person).”